lib/rides/pages/search_suggestion_page.dart

```diff
@@ -173,37 +180,31 @@ class _SearchSuggestionPage extends State<SearchSuggestionPage> {
   }
 
   Widget buildDatePicker() {
-    _dateController.text = localeManager.formatDate(widget.date);
+    _dateController.text = localeManager.formatDate(_selectedDate);
 
     return TextFormField(
       decoration: InputDecoration(
         border: const OutlineInputBorder(),
         labelText: S.of(context).formDate,
       ),
       readOnly: true,
-      onTap: () {
-        _showDatePicker;
-        loadRides();
-      },
+      onTap: _showDatePicker,
       controller: _dateController,
     );
   }
 
   Widget buildTimePicker() {
-    _timeController.text = localeManager.formatTime(widget.date);
+    _timeController.text = localeManager.formatTime(_selectedDate);
 
     return TextFormField(
       decoration: InputDecoration(
         border: const OutlineInputBorder(),
         labelText: S.of(context).formTime,
+        errorText: _timeValidator(_timeController.text),
```

Comment:
why does this work instead of `validator: _timeValidator`?

Reply:
because there is no Form around it so the validator never gets called

Reply:
Ah yes, that makes sense! But why isn't there a form on that page in the first place? because it's never submitted?

Reply:
yes i think so you have to ask @Showcas for that. But it works like this.